expertsystem/amplitude/model.py

```diff
@@ -0,0 +1,349 @@
+"""Data objects for an `~expertsystem.amplitude` model."""
+
+
+from abc import ABC
+from collections import abc
+from dataclasses import dataclass, field
+from enum import Enum, auto
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Iterable,
+    Iterator,
+    List,
+    Optional,
+    Sequence,
+    Set,
+)
+
+from expertsystem.data import Particle, ParticleCollection
+
+
+@dataclass
+class FitParameter(abc.Hashable):
+    name: str
+    value: float = 0.0
+    is_fixed: bool = False
+
+    def __hash__(self) -> int:
+        return hash(str(self))
```

Comment:
there is a `hash=True` option in the `dataclass` decorator. Why is hashable required in the first place?

Reply:
Ah yes 8687a6e5